wakeonlan.py

```diff
@@ -27,8 +27,14 @@ def create_magic_packet(macaddress: str) -> bytes:
     if len(macaddress) == 17:
         sep = macaddress[2]
         macaddress = macaddress.replace(sep, "")
-    elif len(macaddress) != 12:
+    elif len(macaddress) == 14:
+        sep = macaddress[4]
+        macaddress = macaddress.replace(sep, "")
+    if len(macaddress) != 12:
         raise ValueError("Incorrect MAC address format")
+    for s in macaddress:
+        if s not in "0123456789abcdefABCDEF":
+            raise ValueError("Invalid character in the MAC address")
 
     return bytes.fromhex("F" * 12 + macaddress * 16)
```

Comment:
This is extraneous logic. The `bytes.fromhex()` line below will raise a similar error:

```python
>>> mac = 'abcdefghijkl'
>>> bytes.fromhex('F' * 12 + mac * 16)
ValueError: non-hexadecimal number found in fromhex() arg at position 18
```

If there is value in displaying a more specific error, maybe the `return` line should be wrapped in a `try`/`except` block?

```suggestion
try:
    return bytes.fromhex("F" * 12 + macaddress * 16)
except ValueError:
    raise ValueError(f"Invalid character in MAC address: {macaddress}")
```